### PR TITLE
move snapshot and fund PUT endpoints to admin route

### PR DESCRIPTION
**Pull request by enzoc4**

### CODE REVIEW

File: vit-servicing-station-lib/src/v0/endpoints/funds/routes.rs [hunk 1]
```diff
@@ -23,18 +23,21 @@ pub async fn filter(
...
+    // fund_by_id need to be checked first otherwise requests are swallowed by the fund::any
+    root.and(fund_by_id.or(fund).or(all_funds)).boxed()
+}
+
+pub fn admin_filter(
+    context: SharedContext,
+) -> impl Filter<Extract = (impl Reply,), Error = Rejection> + Clone {
+    let with_context = warp::any().map(move || context.clone());
+
+    warp::put()
```
Comment (Zeegomo) [inline vit-servicing-station-lib/src/v0/endpoints/funds/routes.rs:39]: nit: could use `path::end()` here? (not sure if it makes much sense though, just a though)

File: doc/api/v0.yaml [hunk 2]
```diff
@@ -270,6 +253,26 @@ paths:
         "200":
           description: Success
 
+  /api/v0/admin/fund/{id}:
```
[the remark this answers is not in the document]
Comment (enzoc4) [reply doc/api/v0.yaml:256]: actually no, good catch

File: vit-servicing-station-lib/src/v0/endpoints/funds/routes.rs [hunk 3]
```diff
@@ -23,18 +23,21 @@ pub async fn filter(
...
     let all_funds = warp::path::end()
         .and(warp::get())
-        .and(with_context.clone())
+        .and(with_context)
         .and_then(get_all_funds)
         .boxed();
 
-    let put_fund = warp::path::end()
-        .and(warp::put())
+    // fund_by_id need to be checked first otherwise requests are swallowed by the fund::any
```
[the remark this answers is not in the document]
Comment (enzoc4) [reply vit-servicing-station-lib/src/v0/endpoints/funds/routes.rs:30]: That comment isn't mine actually (it's Daniel's I think), it was already there... But yeah, I don't understand it either. Maybe I should remove it?

File: snapshot-service/src/routes.rs [hunk 4]
```diff
@@ -30,18 +30,14 @@ pub fn filter(
 }
 
 pub fn update_filter(
-    root: BoxedFilter<()>,
     context: UpdateHandle,
-) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+) -> impl Filter<Extract = (impl Reply,), Error = Rejection> + Clone {
```
[the remark this answers is not in the document]
Comment (enzoc4) [reply snapshot-service/src/routes.rs:34]: It is the signature for `and`. I just think it's a bit weird to pass a filter as a parameter, when filters can always be composed, so I removed the argument and changed the signature so this compiles

```rust
        let fund_filter = warp::path!("fund" / ..).and(funds::admin_filter(context.clone()));
```

https://docs.rs/warp/latest/warp/trait.Filter.html#extracting-tuples

**Review by Zeegomo — APPROVED** [review]

**Pull request merged**